tesseract_kinematics/kdl/include/tesseract_kinematics/kdl/kdl_utils.h

```diff
@@ -102,6 +102,22 @@ struct KDLChainData
   std::vector<std::pair<std::string, std::string>> chains; /**< The chains used to create the object */
 };
 
+/**
+ * @brief The KDLConfig struct
+ *
+ * This contains parameters that can be used to customize the KDL solver for your application.
+ * They are ultimately passed to the constuctor of the undelying ChainIkSolver.
+ *
+ * The defaults provided here are the same defaults imposed by the KDL library.
+ */
+struct KDLConfig
+{
+  Eigen::VectorXd weights = Eigen::VectorXd::Constant(6, 1.0);
```

Comment:
I would also use a fixed size like what was done in KDL. Also will need to add the Eigen align macro to this struct.
```  c++
Eigen::Matrix< double, 6, 1 >
```